server/routerlicious/packages/routerlicious-base/src/alfred/routes/api/api.ts

```diff
@@ -129,6 +135,20 @@ export function create(
 		},
 	);
 
+	router.post("/broadcast-signal", (req, res) => {
+		try {
+			const tenantId = req.body?.tenantId;
+			const documentId = req.body?.documentId;
```

Comment:
For documentId and tenantId, we need to validate that these match the JWT claims. Otherwise, anyone with a valid tenant could perform operations against any tenant/document, which is a massive problem.

Look at `verifyStorageToken` for an example of how to do this. Unfortunately, that middleware only currently accepts tenantId and documentId that are available in the route params (/:tenantId/:id in the route path). You can easily extend it to take an optional param that allows for different retrieval of tenantId and documentId from the request. Usage would look like this:

```ts
verifyStorageToken(tenantManager, config, {
    getTenantId: (req: Request) => req.body?.tenantId,
    getDocumentId: (req: Request) => req.body?.documentId,
})
```

You would need to implement the getTenantId and getDocumentId options in the middleware definition in `services-utils`

Reply:
`verifyRequest` from this file also functions similarly, but isn't as widely used, and it will throw a 500 for what should be a `401/403` error, so I would still recommend modifying and using `verifyStorageToken`